### DISCUSSION TITLE
How to initalize Firestore with custom Settings like ignoreUndefinedProperties: true

### DISCUSSION THREAD

**Discussion by sugoidesune** [vuejs/vuefire#1540] [Questions and Help]
The traditional way to do this is as follows, but it doesn't work as vuefire already initialized firestore with different settings.

```js
const app = initializeApp(config)
initializeFirestore(app, {
    ignoreUndefinedProperties: true,
})
const db = getFirestore(app);
```

**Comment by szulcus** [comment]
I found some information that might be useful here here: https://github.com/vuejs/vuefire/issues/154, https://vuefire.vuejs.org/guide/other-firebase-services.html#Nuxt.

Currently, in my ~/plugins/firestore.client.ts, I have the following code:
```ts
import { initializeFirestore } from 'firebase/firestore';

export default defineNuxtPlugin(() => {
	const firebaseApp = useFirebaseApp();
	const firestore = initializeFirestore(firebaseApp, {
		experimentalForceLongPolling: true,
	});

	return {
		provide: {
			firestore,
		},
	};
});

```
And use it as `const { $firestore } = useNuxtApp()`, but I'm still waiting for a better solution. Could you provide any guidance on this, @posva? 🙏

**Comment by posva** [comment]
Without Nuxt, you can simply use the Firebase SDK to setup anything needed directly. Since it automatically initializes the first time it's used (e.g. in a component), do it before mounting.

In Nuxt, it cannot be achieved well yet because this feature was added later and there haven't been time to be added yet

**Reply by charlescurrier** [reply]
Is there any issue you're aware of with using the method explained by @szulcus above? I've experimented with using that approach to ignore undefined and configure caching. Both seem to function at first glance.
